src/model/CardMapping.ts

```diff
@@ -4,7 +4,7 @@ class CardMapping {
   public map: Map<CardType, number>
 
   constructor (map?: Map<CardType, number>) {
-    if (map === null) {
+    if (map !== undefined) {
```

Comment:
That works! Another idea would be to use Default parameters: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Functions/Default_parameters